persistence/src/test/java/cz/muni/fi/pa165/mushrooms/dao/MushroomHunterDaoTest.java

```diff
@@ -0,0 +1,84 @@
+package cz.muni.fi.pa165.mushrooms.dao;
+
+import cz.muni.fi.pa165.mushrooms.entity.MushroomHunter;
+import cz.muni.fi.pa165.mushrooms.validation.PersistenceSampleApplicationContext;
+import org.junit.Before;
+import org.junit.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.test.context.ContextConfiguration;
+import org.springframework.test.context.TestExecutionListeners;
+import org.springframework.test.context.junit4.AbstractJUnit4SpringContextTests;
+import org.springframework.test.context.transaction.TransactionalTestExecutionListener;
+import org.springframework.transaction.annotation.Transactional;
+
+import javax.persistence.EntityManager;
+import javax.persistence.PersistenceContext;
+
+import static org.assertj.core.api.Assertions.*;
+
+/**
+ * @author bkompis
+ */
+@ContextConfiguration(classes= PersistenceSampleApplicationContext.class)
+@TestExecutionListeners(TransactionalTestExecutionListener.class)
+@Transactional
+public class MushroomHunterDaoTest extends AbstractJUnit4SpringContextTests {
+
+    @Autowired
+    private MushroomHunterDao mushroomHunterDao;
+
+    @PersistenceContext
+    private EntityManager em;
+
+    private MushroomHunter hunter1;
+    private MushroomHunter hunter2;
+
+    @Before
+    public void setUp(){
+        // create hunters
+        hunter1 = createMushroomHunter("John", "Doe");
+        hunter2 = createMushroomHunter("Jack", "Daniels");
+
+        // persist hunters
+        mushroomHunterDao.create(hunter1);
+        assertThat(mushroomHunterDao.findAll().contains(hunter1));
+        mushroomHunterDao.create(hunter2);
+        assertThat(mushroomHunterDao.findAll().contains(hunter2));
+    }
+

```

Comment:
#I dont like this for 2 reasons:

1) you are using `mushroomHunterDao.create()` in `setUp()` and then testing create in another unit test

2) I feel like assertions should not appear in setup. See [this](https://stackoverflow.com/questions/1371940/unit-testing-is-it-a-good-practice-to-have-assertions-in-setup-methods)

But I have no idea how to fix it. Maybe create() test should run first, but this is breaking execution order independence required in unit tests